src/overlays/actors/ovl_En_Raf/z_en_raf.c

```diff
@@ -80,65 +178,787 @@ static DamageTable D_80A1939C = {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Raf/func_80A18A90.s")
+static Vec3f sMiddleSegmentTargetScaleDuringSpit[] = {
+    { 1.0f, 1.5f, 0.7f }, { 1.0f, 2.0f, 1.5f }, { 1.0f, 2.0f, 0.5f }, { 1.0f, 2.0f, 0.5f }, { 1.0f, 1.0f, 1.0f },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Raf/func_80A18B8C.s")
+static Vec3f sUpperSegmentTargetScaleDuringSpit[] = {
+    { 1.5f, 1.5f, 1.7f }, { 1.5f, 1.5f, 1.3f }, { 3.0f, 1.0f, 0.5f }, { 1.0f, 1.0f, 0.5f }, { 1.0f, 1.0f, 1.0f },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Raf/func_80A18DA0.s")
+void EnRaf_TransformLimbDraw(GlobalContext* globalCtx2, s32 limbIndex, Actor* thisx) {
+    GlobalContext* globalCtx = globalCtx2;
+    EnRaf* this = THIS;
+    s32 i;
+
+    switch (this->petalScaleType) {
+        case EN_RAF_PETAL_SCALE_TYPE_GRAB:
+            if ((limbIndex == CARNIVOROUS_LILY_PAD_LIMB_TRAP_1_MIDDLE_SEGMENT) ||
+                (limbIndex == CARNIVOROUS_LILY_PAD_LIMB_TRAP_3_MIDDLE_SEGMENT) ||
+                (limbIndex == CARNIVOROUS_LILY_PAD_LIMB_TRAP_2_MIDDLE_SEGMENT)) {
+                for (i = 0; i < ARRAY_COUNT(sGrabAnimationCheckFrames); i++) {
+                    if ((s16)this->skelAnime.curFrame == sGrabAnimationCheckFrames[i]) {
+                        Math_Vec3f_Copy(&this->targetLimbScale[limbIndex], &sMiddleSegmentTargetScaleDuringGrab[i]);
+                    }
+                }
+            }
+
+            if ((limbIndex == CARNIVOROUS_LILY_PAD_LIMB_TRAP_1_UPPER_SEGMENT) ||
+                (limbIndex == CARNIVOROUS_LILY_PAD_LIMB_TRAP_3_UPPER_SEGMENT) ||
+                (limbIndex == CARNIVOROUS_LILY_PAD_LIMB_TRAP_2_UPPER_SEGMENT)) {
+                for (i = 0; i < ARRAY_COUNT(sGrabAnimationCheckFrames); i++) {
+                    if ((s16)this->skelAnime.curFrame == sGrabAnimationCheckFrames[i]) {
+                        Math_Vec3f_Copy(&this->targetLimbScale[limbIndex], &sUpperSegmentTargetScaleDuringGrab[i]);
+                    }
+                }
+            }
+            break;
+
+        case EN_RAF_PETAL_SCALE_TYPE_CHEW:
+            if ((limbIndex == CARNIVOROUS_LILY_PAD_LIMB_TRAP_1_MIDDLE_SEGMENT) ||
+                (limbIndex == CARNIVOROUS_LILY_PAD_LIMB_TRAP_3_MIDDLE_SEGMENT) ||
+                (limbIndex == CARNIVOROUS_LILY_PAD_LIMB_TRAP_2_MIDDLE_SEGMENT)) {
+                Math_Vec3f_Copy(&this->targetLimbScale[limbIndex], &sMiddleSegmentTargetScaleDuringGrab[2]);
+            } else if ((limbIndex == CARNIVOROUS_LILY_PAD_LIMB_TRAP_1_UPPER_SEGMENT) ||
+                       (limbIndex == CARNIVOROUS_LILY_PAD_LIMB_TRAP_3_UPPER_SEGMENT) ||
+                       (limbIndex == CARNIVOROUS_LILY_PAD_LIMB_TRAP_2_UPPER_SEGMENT)) {
+                Math_Vec3f_Copy(&this->targetLimbScale[limbIndex], &sUpperSegmentTargetScaleDuringGrab[2]);
+            }
+
+            // These matrix operations make the trap petals look a bit more "wobbly" as it chews.
```

Comment:
I know this was discussed in Discord, I think it essentially stretches them a bit in a particular direction, right? it undoes all the limb rotations, scales/stretches, then puts them all back to get it in the same place.

Reply:
Yeah, that's what it does. I'll make the comment a little more clear as to what exactly is happening here.